@xen-orchestra/vmware-explorer/esxi.mjs

```diff
@@ -0,0 +1,259 @@
+import { Client } from 'node-vsphere-soap'
+import { dirname } from 'node:path'
+import { EventEmitter } from 'node:events'
+import { strictEqual } from 'node:assert'
+import fetch from 'node-fetch'
+
+import parseVmdk from './parsers/vmdk.mjs'
+import parseVmsd from './parsers/vmsd.mjs'
+import parseVmx from './parsers/vmx.mjs'
+
+const MAX_SCSI = 9
+const MAX_ETHERNET = 9
+
+export default class Esxi extends EventEmitter {
+  #client
+  #cookies
+  #host
+  #user
+  #password
+  #ready = false
+
+  constructor(host, user, password, sslVerify = true) {
+    super()
+    this.#host = host
+    this.#user = user
+    this.#password = password
+    // @FIXME : this will emit a warning at the process level is sslVerifiy is false
```

Comment:
```suggestion
    // @FIXME this module inject NODE_TLS_REJECT_UNAUTHORIZED into the process env, which is problematic because it disables globally SSL certificate verification
    //
    // we need to find a fix for this, maybe forking the library
```